File: frontend/src/types/reservation.ts
export type ReservationStatus =
  | "PENDING"
  | "CONFIRMED"
  | "CANCELLED";

export interface RestaurantTable {
  id: number;
  tableNumber: number;
}

export interface ReservationRequest {
  date: string;
  startTime: string;
  endTime: string;
  tableId: number;
  customerName: string;
  phoneNumber: string;
}

export interface ReservationResponse {
  id: number;
  reservationReference: string;
  date: string;
  startTime: string;
  endTime: string;
  tableId: number;
  tableNumber: number;
  customerName: string;
  phoneNumber: string;
  status: ReservationStatus;
  createdAt: string;
}

File: order_service/src/main/java/com/dineflow/order/model/Reservation.java
package com.dineflow.order.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;

import com.dineflow.order.model.enums.ReservationStatus;

import java.time.LocalDateTime;

@Entity
@Table(name = "reservations")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String reservationReference;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false)
    private LocalTime startTime;

    @Column(nullable = false)
    private LocalTime endTime;

    @Column(nullable = false, length = 100)
    private String customerName;

    @Column(nullable = false, length = 20)
    private String phoneNumber;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReservationStatus status;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "table_id", nullable = false)
    private RestaurantTable table;
}